smartlearner/optimizers/adam.py

```diff
@@ -0,0 +1,43 @@
+from collections import OrderedDict
+
+import theano
+import theano.tensor as T
+import numpy as np
+
+from ..interfaces import Optimizer
+from ..utils import sharedX
+
+class Adam(Optimizer):
```

Comment:
I think it'd have been clearer to directly use `-gparam` in `AdaGrad`. In any cases, it'd make the code for Adam much more complex than it needs to be.